mne_bids/path.py

```diff
@@ -310,6 +316,17 @@ def _check(self, deep=True):
                                      f'{ALLOWED_FILENAME_KINDS}.')
 
 
+def _convert_str_to_BIDSPath(bids_basename):
+    """Convert string of bids basename to BIDSPath object."""
+    params = get_entities_from_fname(bids_basename)
+    if params['subject'] == 'emptyroom':
+        check = False
```

Comment:
Could you give me an example why this is needed? I must be missing something…

Reply:
If I comment this out, I get this error:

```
__________________________________ test_get_matched_empty_room __________________________________

    @pytest.mark.filterwarnings(warning_str['meas_date_set_to_none'])
    @pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
    def test_get_matched_empty_room():
        """Test reading of empty room data."""
        bids_root = _TempDir()
    
        raw = _read_raw_fif(raw_fname)
    
        bids_basename = BIDSPath(subject='01', session='01',
                                 task='audiovisual', run='01')
        write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
    
        er_basename = get_matched_empty_room(bids_basename=bids_basename,
                                             bids_root=bids_root)
        assert er_basename is None
    
        # testing data has no noise recording, so save the actual data
        # as if it were noise
        er_raw_fname = op.join(data_path, 'MEG', 'sample', 'ernoise_raw.fif')
        raw.crop(0, 10).save(er_raw_fname, overwrite=True)
    
        er_raw = _read_raw_fif(er_raw_fname)
        er_date = er_raw.info['meas_date']
        if not isinstance(er_date, datetime):
            # mne < v0.20
            er_date = datetime.fromtimestamp(er_raw.info['meas_date'][0])
        er_date = er_date.strftime('%Y%m%d')
        er_bids_basename = BIDSPath(subject='emptyroom', task='noise',
                                    session=er_date, kind='meg', check=False)
>       write_raw_bids(er_raw, er_bids_basename, bids_root, overwrite=True)

bids_basename = BIDSPath(sub-01_ses-01_task-audiovisual_run-01)
bids_root  = '/var/folders/0p/knts5_1j38j6wvv3w_19584r0000gn/T/tmp_mne_tempdir_3kbffnen'
er_basename = None
er_bids_basename = BIDSPath(sub-emptyroom_ses-20021203_task-noise_meg)
er_date    = '20021203'
er_raw     = <Raw | ernoise_raw.fif, 376 x 3004 (10.0 s), ~3.6 MB, data not loaded>
er_raw_fname = '/Users/adam2392/mne_data/MNE-testing-data/MEG/sample/ernoise_raw.fif'
raw        = <Raw | sample_audvis_trunc_raw.fif, 376 x 3004 (10.0 s), ~3.6 MB, data not loaded>

mne_bids/tests/test_read.py:588: 
```

which is caused by:

```
    def _check_empty_room_basename(bids_path, on_invalid_er_session='raise',
                                   on_invalid_er_task='raise'):
        if bids_path.task != 'noise':
            msg = (f'task must be "noise" if subject is "emptyroom", but '
                   f'received: {bids_path.task}')
            if on_invalid_er_task == 'raise':
>               raise ValueError(msg)
E               ValueError: task must be "noise" if subject is "emptyroom", but received: None

bids_path  = BIDSPath(/var/folders/0p/knts5_1j38j6wvv3w_19584r0000gn/T/tmp_mne_tempdir_3kbffnen/sub-emptyroom/ses-20021203/sub-emptyroom_ses-20021203_scans.tsv)
msg        = 'task must be "noise" if subject is "emptyroom", but received: None'
on_invalid_er_session = 'raise'
on_invalid_er_task = 'raise'

mne_bids/utils.py:340: ValueError
```

Reply:
I'm assuming it's because the `*_scans.tsv` doesn't have a task entity, but it gets hit w/ the emptyroom check.

Reply:
jk that check is a bug on our end, which can be fixed (https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/02-magnetoencephalography.html#empty-room-meg-recordings). It seems like you should not check the scans file for the `task` entity in emptyroom.

Reply:
> It seems like you should not check the scans file for the `task` entity in emptyroom.

The `scans` file **never** has any entities but `sub` and `ses`:

```
    sub-<label>[_ses-<label>]_scans.tsv
```
https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html#scans-file

Reply:
Correct me if I'm wrong, but `emptyroom` subjects should only undergo the task check IF they are the actual recording, or sidecar json right? 

So `coordsystem`, and `electrodes`, etc. don't need to have that check.

Reply:
See: https://github.com/mne-tools/mne-bids/pull/511/files#r475889074